presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -99,49 +132,67 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
-                LabelNode label = new LabelNode("inHash" + bucket.getKey());
-                switchBuilder.addCase(bucket.getKey(), label);
-                Collection<ByteCodeNode> testValues = bucket.getValue();
-
-                ByteCodeBlock caseBlock = buildInCase(generatorContext, scope, type, label, match, defaultLabel, testValues, false);
-                switchCaseBlocks
-                        .append(caseBlock.setDescription("case " + bucket.getKey()));
-            }
-            switchBuilder.defaultCase(defaultLabel);
-
-            Binding hashCodeBinding = generatorContext
-                    .getCallSiteBinder()
-                    .bind(hashCodeFunction.getMethodHandle());
-
-            switchBlock = new ByteCodeBlock()
-                    .comment("lookupSwitch(hashCode(<stackValue>))")
-                    .dup(javaType)
-                    .append(invoke(hashCodeBinding, hashCodeFunction.getSignature()))
-                    .longToInt()
-                    .append(switchBuilder.build())
-                    .append(switchCaseBlocks);
-        }
-        else {
-            // TODO: replace Set with fastutils (or similar) primitive sets if types are primitive
-            // for huge IN lists, use a Set
-            Binding constant = generatorContext.getCallSiteBinder().bind(constantValues, Set.class);
-
-            switchBlock = new ByteCodeBlock()
-                    .comment("inListSet.contains(<stackValue>)")
-                    .append(new IfStatement()
-                            .condition(new ByteCodeBlock()
-                                    .comment("value (+boxing if necessary)")
-                                    .dup(javaType)
-                                    .append(ByteCodeUtils.boxPrimitive(javaType))
-                                    .comment("set")
-                                    .append(loadConstant(constant))
-                                    // TODO: use invokeVirtual on the set instead. This requires swapping the two elements in the stack
-                                    .invokeStatic(CompilerOperations.class, "in", boolean.class, Object.class, Set.class))
-                            .ifTrue(jump(match)));
+        ByteCodeBlock switchCaseBlocks = new ByteCodeBlock();
+        LookupSwitch.LookupSwitchBuilder switchBuilder = lookupSwitchBuilder();
+        switch (switchGenerationCase) {
+            case SMALL_LIST_SMALL_PRIMITIVES:
+                int[] constantValuesArray = new int[constantValues.size()];
+                int i = 0;
+                for (Object object : constantValues) {
+                    constantValuesArray[i++] = ((Long) object).intValue();
+                }
+                Arrays.sort(constantValuesArray);
+                for (int constantValue : constantValuesArray) {
+                    switchBuilder.addCase(constantValue, match);
+                }
```

Comment:
You don't need the sort. I understand that I asked you to add it. But it turns out it's sorted by line 81 of `LookupSwitch` as I digged more into it.
```
                for (Object constantValue : constantValues) {
                    switchBuilder.addCase(((Long) constantValue).intValue(), match);
                }
```